### ISSUE TITLE
NotImplementedError: for 2D images

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#2006]
data: 

~~~
sct_register_multimodal -i Sample1_mask.nii.gz -o Sample1_mask_reg.nii.gz -d ref_template_50um_Sample1.nii.gz -identity 1

--
Spinal Cord Toolbox (master/925987e5b179a67cdc95ea9bc10aa7e4758c69d2)
Running /Users/julien/code/sct/scripts/sct_register_multimodal.py -i Sample1_mask.nii.gz -o Sample1_mask_reg.nii.gz -d ref_template_50um_Sample1.nii.gz -identity 1

Input parameters:
  Source .............. Sample1_mask.nii.gz
  Destination ......... ref_template_50um_Sample1.nii.gz
  Init transfo ........ 
  Mask ................ 
  Output name ......... Sample1_mask_reg.nii.gz
  Remove temp files ... 1
  Verbose ............. 1

Check if input data are 3D...

Create temporary folder (/var/folders/6f/wy6ljmx9453cgth2qwv5l1s80000gn/T/sct-20180821154559.897919-rtDRtW)...

Copying input data to tmp folder and convert to nii...
Traceback (most recent call last):
  File "/Users/julien/code/sct/scripts/sct_register_multimodal.py", line 802, in <module>
    main()
  File "/Users/julien/code/sct/scripts/sct_register_multimodal.py", line 432, in main
    Image('dest.nii').change_orientation("RPI").save('dest_RPI.nii')
  File "/Users/julien/code/sct/spinalcordtoolbox/image.py", line 369, in change_orientation
    change_orientation(self, orientation, self, inverse=inverse)
  File "/Users/julien/code/sct/spinalcordtoolbox/image.py", line 897, in change_orientation
    raise NotImplementedError("Don't know how to change orientation for this image")
NotImplementedError: Don't know how to change orientation for this image
~~~

**Comment by zougloub** [comment]
What is the expected behavior in this case?

**Comment by jcohenadad** [comment]
expected behavior would be to interpolate the data to be in the destination space (ref_template_50um_Sample1). It was working on v3.1.1.

**Comment by zougloub** [comment]
Will reshape to 3D and see if we can get away with it. Haven't looked at the problem yet.

**Comment by zougloub** [comment]
@jcohenadad I don't have your data, so if you can quickly test the branch from PR #2004 on this, I'd be thankful.

**Comment by jcohenadad** [comment]
i'm now getting this:
~~~
sct_register_multimodal -i Sample1_mask.nii.gz -o Sample1_mask_reg.nii.gz -d ref_template_50um_Sample1.nii.gz -identity 1

--
Spinal Cord Toolbox (cJ/2002-typo/39b1047c13e1d951b52b8c8864b278ba2f721abf)
Running /Users/julien/code/sct/scripts/sct_register_multimodal.py -i Sample1_mask.nii.gz -o Sample1_mask_reg.nii.gz -d ref_template_50um_Sample1.nii.gz -identity 1

Input parameters:
  Source .............. Sample1_mask.nii.gz
  Destination ......... ref_template_50um_Sample1.nii.gz
  Init transfo ........ 
  Mask ................ 
  Output name ......... Sample1_mask_reg.nii.gz
  Remove temp files ... 1
  Verbose ............. 1

Check if input data are 3D...

Create temporary folder (/var/folders/6f/wy6ljmx9453cgth2qwv5l1s80000gn/T/sct-20180821233824.574041-PPfyUs)...

Copying input data to tmp folder and convert to nii...
Traceback (most recent call last):
  File "/Users/julien/code/sct/scripts/sct_register_multimodal.py", line 802, in <module>
    main()
  File "/Users/julien/code/sct/scripts/sct_register_multimodal.py", line 432, in main
    Image('dest.nii').change_orientation("RPI").save('dest_RPI.nii')
  File "/Users/julien/code/sct/spinalcordtoolbox/image.py", line 369, in change_orientation
    change_orientation(self, orientation, self, inverse=inverse)
  File "/Users/julien/code/sct/spinalcordtoolbox/image.py", line 923, in change_orientation
    data = data[::inversion[0], ::inversion[1], ::inversion[2]]
IndexError: too many indices for array
~~~

the funky data are accessible [here](https://www.dropbox.com/s/0vpk2jrpol9lsyh/20180821_data_that_will_give_you_insomnia.zip?dl=0).

**Comment by zougloub** [comment]
Now not crashing with another fix in image. Not sure if the result is the expected one though.

**Comment by jcohenadad** [comment]
it is the expected one (i.e. same as with v3.1.1). Thanks!
Fixed via #2004
